src/client/Microsoft.Identity.Client.Brokers/RuntimeBroker.cs

```diff
@@ -0,0 +1,392 @@
+        private const string WamErrorPrefix = "WAM Error ";
+
+        /// <summary>
+        /// Ctor. Only call if on Win10, otherwise a TypeLoadException occurs. See DesktopOsHelper.IsWin10
+        /// </summary>
+        public RuntimeBroker(
+            CoreUIParent uiParent,
+            ApplicationConfiguration appConfig,
+            ICoreLogger logger)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+
+            _parentHandle = GetParentWindow(uiParent);
+
+            _wamOptions = appConfig.WindowsBrokerOptions ??
+                WindowsBrokerOptions.CreateDefault();
+
+            if (_wamOptions.ListWindowsWorkAndSchoolAccounts)
+            {
+                throw new NotImplementedException("The new broker implementation does not yet support Windows account discovery (ListWindowsWorkAndSchoolAccounts option)");
+            }
+        }
+
+        /// <summary>
+        /// Acquire Token Interactively 
+        /// </summary>
+        /// <param name="authenticationRequestParameters"></param>
+        /// <param name="acquireTokenInteractiveParameters"></param>
+        /// <returns></returns>
+        /// <exception cref="MsalServiceException"></exception>
+        public async Task<MsalTokenResponse> AcquireTokenInteractiveAsync(
+            AuthenticationRequestParameters authenticationRequestParameters,
+            AcquireTokenInteractiveParameters acquireTokenInteractiveParameters)
+        {
+            MsalTokenResponse msalTokenResponse = null;
+
+            //need to provide a handle
+            if (_parentHandle == IntPtr.Zero)
+            {
+                throw new MsalClientException(
+                    "window_handle_required",
+                    "Public Client applications wanting to use WAM need to provide their window handle. Console applications can use GetConsoleWindow Windows API for this.");
+            }
+
+            //if OperatingSystemAccount is passed then we use the user signed-in on the machine
+            if (PublicClientApplication.IsOperatingSystemAccount(authenticationRequestParameters.Account))
+            {
+                return await AcquireTokenInteractiveDefaultUserAsync(authenticationRequestParameters, acquireTokenInteractiveParameters).ConfigureAwait(false);
+            }
+
+            var cancellationToken = authenticationRequestParameters.RequestContext.UserCancellationToken;
+                
+            _logger.Verbose("[WamBroker] Using Windows account picker.");
+
+            using (var core = new NativeInterop.Core())
+            using (var authParams = WamAdapters.GetCommonAuthParameters(authenticationRequestParameters, _wamOptions.MsaPassthrough))
+            {
+                //Login Hint
+                string loginHint = authenticationRequestParameters.LoginHint ?? authenticationRequestParameters?.Account?.Username;
+
+                if (!string.IsNullOrEmpty(loginHint))
```

Comment:
This block is a bit verbose, maybe a single logging statement is enough, something like

`
_logger.Verbose("[WamBroker] AcquireTokenInteractive - login hint provided? " + string.isnullorempty(_loginHint));
`